subsys/net/openthread/Kconfig

```diff
@@ -19,12 +19,46 @@ config NORDIC_SECURITY_PROMPTLESS
 choice OPENTHREAD_SECURITY
 	depends on !OPENTHREAD_SECURITY_INTERNAL
 
+config OPENTHREAD_NRF_SECURITY_PSA_CHOICE
+	bool "nRF Security with PSA crypto enabled"
+	select OPENTHREAD_NRF_SECURITY_PSA
+
 config OPENTHREAD_NRF_SECURITY_CHOICE
 	bool "nRF Security"
 	select OPENTHREAD_NRF_SECURITY
 
 endchoice
 
+config OPENTHREAD_NRF_SECURITY_PSA
+	bool
+	select EXPERIMENTAL
+	select NRF_SECURITY
+	select MBEDTLS_ENABLE_HEAP
+	select MBEDTLS_TLS_LIBRARY
+	select MBEDTLS_PSA_CRYPTO_STORAGE_C if !BUILD_WITH_TFM
+	select PSA_NATIVE_ITS if !BUILD_WITH_TFM
+	select PSA_WANT_ALG_HKDF
+	select PSA_WANT_ALG_CCM
+	select PSA_WANT_ALG_CMAC
+	select PSA_WANT_ALG_HMAC
+	select PSA_WANT_ALG_ECB_NO_PADDING
+	select PSA_WANT_ALG_SHA_256
+	select PSA_WANT_ALG_JPAKE
+	select PSA_WANT_ALG_TLS12_ECJPAKE_TO_PMS
+	select PSA_WANT_ALG_DETERMINISTIC_ECDSA if OPENTHREAD_ECDSA
```

Comment:
You could also do it like this:

```
config OPENTHREAD_NRF_SECURITY_PSA
    bool
    select PSA_<basic_features>

config OPENTHREAD_NRF_SECURITY_PSA
     select PSA_WANT_<ecdsa_features>
     depends on OPENTHREAD_ECDSA
```

Reply:
I agree, but as above- the dependency between OPENTHREAD_ECDSA and functionalities like Joiner which requires ECJPAKE are not directly dependent now. I think we should fix that first and then optimize this part accordingly?

Reply:
Fair enough, it's only a suggestion anyway.